pkg/validation/app.go

```diff
@@ -256,6 +257,22 @@ func (v *Validator) validateKubeConfig(ctx context.Context, cr v1alpha1.App) err
 }
 
 func (v *Validator) validateLabels(ctx context.Context, cr v1alpha1.App) error {
+	// This is for migrations towards managing App CRs from the org namespace.
+	// For the transition time being we must retain backward compatibility for the
+	// App CRs in cluster namespaces.
+	isManagedInOrg := !key.InCluster(cr) && key.IsManagedInOrg(cr)
+
+	var validationMethod func(context.Context, v1alpha1.App) error
+	if isManagedInOrg {
+		validationMethod = v.validateLabelsOrgApp
+	} else {
+		validationMethod = v.validateLablesClusterApp
+	}
+
+	return validationMethod(ctx, cr)
+}
+
+func (v *Validator) validateLablesClusterApp(ctx context.Context, cr v1alpha1.App) error {
```

Comment:
```suggestion
func (v *Validator) validateClusterLabels(ctx context.Context, cr v1alpha1.App) error {
```

I don't think we need app in the name as this package is for app validation.

Reply:
Makes sense